### PR TITLE
ACM-34843: Implement new NodePool health statuses

### PR DESCRIPTION
**Pull request by fxiang1**
## Summary

Better propagate NodePool status to the UI to save users time from having to chase down the resource YAML. Previously the console only checked the `Ready` condition and showed "Ready" or "Pending". All other conditions (errors, warnings, degraded states) were invisible in the UI.

Jira: [ACM-34843](https://redhat.atlassian.net/browse/ACM-34843)

## Acceptance Criteria Checklist

- [x] All HyperShift NodePool condition types are handled (not just `Ready`)
- [x] Error conditions show red error status with meaningful text derived from condition reason/message
- [x] Warning conditions show yellow warning status
- [x] Updating conditions show in-progress/updating status
- [x] Condition details (message, reason) are accessible via popover on the status cell
- [x] The two duplicate status functions are consolidated into one
- [x] NodePool condition type constants are defined (no magic strings)
- [x] Upgrade-readiness gating in DistributionField and ClusterActionDropdown still works correctly
- [x] Unit tests cover all status derivation paths (error, warning, updating, pending, ready)
- [x] Existing tests are updated to reflect the new status logic

## Changeset Summary

| File | Action | Description |
|------|--------|-------------|
| `frontend/src/resources/node-pool.ts` | Modified | Added `NodePoolConditionType` enum (17 conditions), `NodePoolStatus`/`NodePoolStatusType`/`NodePoolCondition` types, and consolidated `getNodePoolStatus()` with priority-based derivation (error > warning > updating > pending > ready) |
| `frontend/src/resources/node-pool.test.ts` | Created | 33 unit tests covering all status derivation paths |
| `frontend/src/routes/.../NodePoolsTable.tsx` | Modified | Replaced simple icon rendering with `AcmInlineStatus` + popover showing all condition details; removed old `getNodepoolStatus` |
| `frontend/src/routes/.../NodePoolsProgress.tsx` | Modified | Replaced old duplicate functions with consolidated function; stepper now reflects error (red) and warning (yellow) states |
| `frontend/src/routes/.../NodePoolsProgress.test.tsx` | Modified | Updated tests for new function signature; added error/warning stepper tests |
| `frontend/src/routes/.../DistributionField.tsx` | Modified | Updated import to consolidated function; uses `.isReady` for upgrade gating |
| `frontend/src/routes/.../ClusterActionDropdown.tsx` | Modified | Updated import to consolidated function; uses `.isReady` for upgrade gating |

## Status Priority Logic

The unified `getNodePoolStatus()` derives status using this priority (highest to lowest):

1. **Error** (red) — Any of `ValidGeneratedPayload`, `ValidReleaseImage`, `ValidMachineConfig`, `ValidArchPlatform`, `ValidPlatformConfig`, `AWSSecurityGroupAvailable`, `ReconciliationActive` has `status: 'False'`
2. **Warning** (yellow) — `AllMachinesReady: False` while `Ready: True` (partial degradation), `AllNodesHealthy: False`, `SupportedVersionSkew: False`, `ReachedIgnitionEndpoint: False`
3. **Updating** (blue) — `UpdatingVersion: True` or `UpdatingConfig: True`
4. **Pending** (grey/spinner) — `Ready` is `False`/`Unknown` and no error conditions
5. **Ready** (green) — `Ready: True` and no error/warning conditions

## Test Results

```
New tests added: 33 (node-pool.test.ts) + 2 (NodePoolsProgress.test.tsx error/warning stepper)
All targeted tests: 93 passed
Full suite: 5522 passed, 3 failed (pre-existing in ApplicationDetails.test.tsx), 7 skipped
Lint: 0 warnings, 0 errors
TypeScript: 0 errors
```

## How to Test (Plugin Mode)

1. `npm run plugins` to start the console in plugin mode
2. Navigate to Infrastructure → Clusters → select a HyperShift hosted cluster → Node pools tab
3. Verify:
   - Ready node pools show green check icon with "Ready" text
   - Clicking the status text opens a popover showing all conditions with icons and messages
   - Error/warning/updating states render with appropriate colors and icons (requires controller-set conditions on real infrastructure)
4. Verify upgrade gating still works:
   - Node pools with non-Ready status should not show "Update available" in the Distribution column

image

## Out of Scope

- Changes to the CIM/agent-based HyperShift flow (`AIHypershiftClusterDetails`)
- Backend/API changes — this is purely a UI improvement consuming existing API data
- Playwright E2E tests — impractical per ticket (conditions are controller-set, not user-configurable)

[ACM-34843]: https://redhat.atlassian.net/browse/ACM-34843?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

* **New Features**
  * Added a unified node pool status model (ok, pending, warning, error, updating) with condition-aware UI.
  * Enhanced node pool UI with inline status popovers that list underlying conditions.

* **Bug Fixes**
  * Improved status precedence (error/warning/updating/ready/pending) and readiness/update detection logic.
  * Updated “update available” checks to use the shared node pool update helper.

* **Tests**
  * Expanded Jest coverage for node pool status edge cases and version comparison behavior.

* **Refactor**
  * Centralized node pool status/status aggregation logic across the UI components.

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/NodePoolsTable.tsx [hunk 1]
```diff
@@ -22,15 +30,69 @@ type NodePoolsTableProps = {
...
-    if (condition.type === 'Ready') {
-      return condition.status === 'True' ? 'Ready' : 'Pending'
-    }
+const STATUS_LABEL_MAP: Record<string, string> = {
+  error: 'Error',
+  warning: 'Warning',
+  updating: 'Updating',
+  pending: 'Pending',
+  ok: 'Ready',
+}
```
[the remark this answers is not in the document]
Comment (fxiang1) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/NodePoolsTable.tsx]: The translation key "Updating" already exists in frontend/public/locales/en/translation.json at line 3580. The t('Updating') call resolves correctly.

**Review by Randy424 — APPROVED** [review]
/lgtm
